modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormTemplateContextFactoryHelper.java

```diff
@@ -135,4 +130,7 @@ protected boolean isDDMFormFieldEvaluable(DDMFormField ddmFormField) {
 		return false;
 	}
 
+	private static final Pattern _functionParametersPattern = Pattern.compile(
+		"'?([\\w]+)'?\\s*[,\\)]\\s*");
```

Comment:
@cleydyr, what is the difference between this regex and the previous one?

Reply:
The previous regex was to match the field name optionally surrounded by apostrophes. The pattern had to be compiled at runtime to include the field name in the pattern itself (more than 3*<number of fields in form>*<number of properties of each field type>).

The new regex is compiled only once per class loading. The regex matches
* the arguments of the function that have alphanumeric characters optionally surrounded by apostrophe;
* after the argument above, follows zero or more whitespace characters;
* after the spaces follows either a comma (if there's another argument following) or a closing parenthesis (if that's the last argument);
* after the comma or the closing parenthesis above, there is zero or more whitespace characters.

I used [capturing group](https://www.regular-expressions.info/brackets.html) so that even though the regex matches with all the criteria above, only the arguments (alphanumeric characters) are captured (no commas, no parenthesis, no white spaces, no apostrophes).

You can get the description and play with the regex at https://regex101.com/r/B2DUzS/1